dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphHostControl.xaml.cs

```diff
@@ -27,37 +27,51 @@
 using SharpDX.Direct2D1;
 using SharpDX.DXGI;
 using SharpDX.Mathematics.Interop;
-using System.Collections.Generic;
+using System;
 using System.Windows;
 using System.Windows.Controls;
 using System.Windows.Media;
-using SysWin = System.Windows;
 using Windows.UI.Composition;
+using SysWin = System.Windows;
 
 
 namespace VisualLayerIntegration
 {
     /// <summary>
     /// Interaction logic for BarGraphHostControl.xaml
     /// </summary>
-    public partial class BarGraphHostControl : UserControl
+    public partial class BarGraphHostControl : UserControl, IDisposable
     {
-        private CompositionHost compositionHost;
-        private Compositor compositor;
-        private Windows.UI.Composition.ContainerVisual graphContainer;
-        private BarGraph currentGraph;
+        private readonly CompositionHost _compositionHost;
+        private readonly Compositor _compositor;
+        private readonly Windows.UI.Composition.ContainerVisual _graphContainer;
+        private BarGraph _currentGraph;
 
-        private double currentDpiX = 96.0;
-        private double currentDpiY = 96.0;
+        private double _currentDpiX = 96.0;
+        private double _currentDpiY = 96.0;
 
-        protected WindowRenderTarget windowRenderTarget;
-        private static RawColor4 white = new RawColor4(255, 255, 255, 255);
+        private WindowRenderTarget _windowRenderTarget;
+        private readonly static RawColor4 _backgroundColor = new RawColor4(255, 255, 255, 255);
+        private readonly Windows.UI.Color[] _graphColors = { Windows.UI.Color.FromArgb(255, 246, 65, 108), Windows.UI.Color.FromArgb(255, 255, 246, 183) };
+        private readonly string _xAxisTitle = "Investment #";
+        private readonly string _yAxisTitle = "# Shares of Stock";
```

Comment:
Order fields: const first, then static before non-static, readonly before non-readonly.
Use "private static readonly RawColor4 ..." instead of "private readonly static RawColor4 ..."
These strings should be const rather than readonly, and should be named either c_xAxisTitle of XAxisTitle.